WordPress/Classes/Services/In-App Update/InAppUpdateCoordinator.swift

```diff
@@ -0,0 +1,143 @@
+                    return .blocking(appStoreInfo)
+                }
+                if currentVersion.isLower(than: appStoreInfo.version) {
+                    return .flexible
+                }
+            }
+
+            return nil
+        }
+    }
+
+    private var currentOsVersion: String {
+        UIDevice.current.systemVersion
+    }
+
+    private var currentVersion: String? {
+        guard let version = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String else {
+            DDLogError("No CFBundleShortVersionString found in Info.plist")
+            return nil
+        }
+        return version
+    }
+
+    private var blockingVersion: String? {
+        let parameter: RemoteConfigParameter = AppConfiguration.isJetpack
+            ? .jetpackInAppUpdateBlockingVersion
+            : .wordPressInAppUpdateBlockingVersion
+        return parameter.value(using: remoteConfigStore)
+    }
+
+    private func fetchAppStoreInfo() async -> AppStoreInfo? {
+        do {
+            let response = try await service.lookup(appID: AppConstants.itunesAppID)
+            return response.results.first { $0.trackId == Int(AppConstants.itunesAppID) }
+        } catch {
+            DDLogError("Error fetching app store info: \(error)")
+            return nil
+        }
+    }
+
+    private func showNotice() {
+        let notice = Notice(
+            title: Strings.Notice.title,
+            message: Strings.Notice.message,
+            feedbackType: .warning,
+            style: InAppUpdateNoticeStyle(),
+            actionTitle: Strings.Notice.update
+        ) { [weak self] _ in
+            self?.openAppStore()
+        }
+        ActionDispatcher.dispatch(NoticeAction.post(notice))
+        // Todo: if the notice is dismissed, show notice again after a defined interval
+    }
+
+    private func showBlockingUpdate(using appStoreInfo: AppStoreInfo) {
+        guard let window = UIApplication.sharedIfAvailable()?.mainWindow,
+              let topViewController = window.topmostPresentedViewController,
+              !((topViewController as? UINavigationController)?.viewControllers.first is BlockingUpdateViewController) else {
+            wpAssertionFailure("Failed to show blocking update view")
+            return
+        }
+        let viewModel = BlockingUpdateViewModel(appStoreInfo: appStoreInfo) { [weak self] in
+            self?.openAppStore()
+        }
+        let controller = BlockingUpdateViewController(viewModel: viewModel)
+        let navigation = UINavigationController(rootViewController: controller)
+        topViewController.present(navigation, animated: true)
+    }
+
+    private func openAppStore() {
+        // Todo
+    }
+}
+
+private enum Strings {
+    enum Notice {
+        static let title = NSLocalizedString("inAppUpdate.notice.title", value: "App Update Available", comment: "Title for notice displayed when there's a newer version of the app available")
```

Comment:
Is there going to be a different copy for a require (blocking) update?

Reply:
Good point, I extracted the the strings to the viewmodel. Both the notice and the modal use the same strings.